File: 백준/Silver/4673. 셀프 넘버/셀프 넘버.py
def d(n):
    return n + sum(map(int, str(n)))
arr = [0] * 10001
for i in range(1, 10001):
    if d(i) < 10001:
        arr[d(i)] = 1
for i in range(1, 10001):
    if arr[i] == 0:
        print(i)

File: 백준/Silver/4673. 셀프 넘버/셀프 넘버.c
#include <stdio.h>

int num(int n) {
	int sum = n;
	for (; n > 0; n /= 10)
		sum += n % 10;
	return sum;
}

main() {
	int arr[10001], i, check;
	for (i = 1; i < 10001; i++) {
		check = num(i);
		if (check < 10001)
			arr[check] = 1;
	}
	for (i = 1; i < 10001; i++)
		if (arr[i] != 1)
			printf("%d\n", i);
	return 0;
}